Export EnvError & EnvMissingError

simonua opened the pull request:
Fixes #93

SimenB commented:
Actually, it seems like this is already supported?

https://github.com/af/envalid/blob/1c3b7a1f92ede28e25bd7e98eb422e938bf80840/src/envalidWithoutDotenv.js#L139-L141

https://runkit.com/embed/2td3jxm0ia9q

Sorry about that 😅 Could you update the readme to properly indicate what's exported?

simonua commented:
@SimenB, I see the export now you are pointing out, but it's not covered in or by the TS definition, is it?

SimenB commented:
Correct, seems to be missing: https://github.com/af/envalid/blob/1c3b7a1f92ede28e25bd7e98eb422e938bf80840/src/envalid.d.ts

af commented:
Ha, completely forgot they were exported 😂

simonua commented:
I think the _envalid.d.ts_ file I modified would remain valid, considered how _envalidWithoutDotenv.d.ts_ exports. What would you say if I reverted the _envalid.js_ then and just stick with the file change for the definition?

SimenB commented:
Perfect! And maybe the readme 🙂

simonua commented:
Will be early next week, but I'll see to it.

simonua commented:
Hi @SimenB & @af,

Have a look, please, and let me know if this is in line with what you expect. Thank you!

af commented:
Looks good to me, thanks! @SimenB In the future, feel free to merge 👍

SimenB commented:
Could you make a patch release as well? We actually use those ourselves with a `@ts-ignore` now 😀

af commented:
4.2.1 is hot off the presses 🔥

On Tue, Feb 5, 2019 at 11:51 AM Simen Bekkhus <notifications@github.com>
wrote:

> Could you make a patch release as well? We actually use those ourselves
> with a @ts-ignore now 😀
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/af/envalid/pull/94#issuecomment-460779018>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AABimRQk9hpqQ8fwyHhq0ePntLV4u2e8ks5vKeCsgaJpZM4ae1C7>
> .
>
